infrastructure/modules/armonik/authentication-in-database.tf

```diff
@@ -0,0 +1,236 @@
+resource "kubernetes_job" "authentication_in_database" {
+  depends_on = [
+    kubernetes_service.ingress
+  ]
+  count = local.authentication_require_authentication ? 1 : 0
+  metadata {
+    name      = "authentication-in-database"
+    namespace = var.namespace
+    labels = {
+      app     = "armonik"
+      service = "authentication-in-database"
+      type    = "monitoring"
+    }
+  }
+  spec {
+    template {
+      metadata {
+        name = "authentication-in-database"
+        labels = {
+          app     = "armonik"
+          service = "authentication-in-database"
+          type    = "monitoring"
+        }
+      }
+      spec {
+        node_selector = local.job_authentication_in_database_node_selector
+        dynamic "toleration" {
+          for_each = (local.job_authentication_in_database_node_selector != {} ? [
+            for index in range(0, length(local.job_authentication_in_database_node_selector_keys)) : {
+              key   = local.job_authentication_in_database_node_selector_keys[index]
+              value = local.job_authentication_in_database_node_selector_values[index]
+            }
+          ] : [])
+          content {
+            key      = toleration.value.key
+            operator = "Equal"
+            value    = toleration.value.value
+            effect   = "NoSchedule"
+          }
+        }
+        dynamic "image_pull_secrets" {
+          for_each = (var.authentication.image_pull_secrets != "" ? [1] : [])
+          content {
+            name = var.authentication.image_pull_secrets
+          }
+        }
+        restart_policy = "OnFailure" # Always, OnFailure, Never
+        container {
+          name              = var.authentication.name
+          image             = var.authentication.tag != "" ? "${var.authentication.image}:${var.authentication.tag}" : var.authentication.image
+          image_pull_policy = var.authentication.image_pull_policy
+          command           = ["/bin/bash", "-c", local.authentication_script]
+          env {
+            name  = "MongoDB_Host"
+            value = local.mongodb_host
+          }
+          env {
+            name  = "MongoDB_Port"
+            value = local.mongodb_port
+          }
+          dynamic "env" {
+            for_each = local.pod_authentication_in_database_credentials
+            content {
+              name = env.key
+              value_from {
+                secret_key_ref {
+                  key      = env.value.key
+                  name     = env.value.name
+                  optional = false
+                }
+              }
+            }
+          }
+          dynamic "volume_mount" {
```

Comment:
You can merge this volume_mount blocks into the following dynamic volume_mount block